src/main/java/Parser.java

```diff
@@ -0,0 +1,208 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+
+import java.io.IOException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalTime;
+
+public class Parser {
+
+    static Scanner sc = new Scanner(System.in);
+    private static UI ui;
+    private static Storage storage;
+    private static int count;
+    private Tasklist tasklist;
+
+    /**
+     * Parser constructor
+     *
+     * @param tasklist tasklist containing the list of tasks
+     * @param ui the ui to be used
+     * @param storage where the tasks will be stored
+     *
+     */
+
+    public Parser(Tasklist tasklist, UI ui, Storage storage) {
+        this.tasklist = tasklist;
+        this.ui = ui;
+        this.storage = storage;
+
+
+        if (tasklist.returnTasks().isEmpty()) {
+            count = 0;
+        } else {
+            count = tasklist.size();
+        }
+    }
+
+    /**
+     * Reads the user input and converts it to instructions.
+     *
+     * @throws IOException  if the user input is wrong
+     */
+
+    public void parserRead() throws IOException {
+        while (sc.hasNext()) {
+            String command = sc.nextLine();
+            if (command.equals("bye")) {
+                UI.bye();
```

Comment:
Is calling static functions directly from the UI class advisable? What if we wanted a ui change?